stablehlo/reference/Element.cpp

```diff
@@ -538,6 +564,38 @@ Element min(const Element &e1, const Element &e2) {
       });
 }
 
+Element power(const Element &e1, const Element &e2) {
+  Type type = e1.getType();
+
+  if (isSupportedIntegerType(type)) {
+    APInt base = e1.getIntegerValue();
+    APInt exponent = e2.getIntegerValue();
+    bool isSigned = base.isSignedIntN(base.getBitWidth());
+    if (isSigned && exponent.isNegative()) {
+      if (base.isZero())
+        llvm::report_fatal_error("divide by zero unsupported");
+      else if (base.abs().isOne())
+        exponent = exponent.abs();
+      else
+        return Element(type, APInt(base.getBitWidth(), 0, isSigned));
+    }
+    const APInt kOne(base.getBitWidth(), 1, isSigned);
+    APInt result(kOne);
+    while (!exponent.isZero()) {
+      if ((exponent & kOne).getBoolValue()) result *= base;
+      base *= base;
+      exponent = exponent.lshr(1);
+    }
```

Comment:
Proposed refactoring:

```
APInt result(base.getBitWidth(), 1, isSigned);
while (!exponent.isZero()) {
  if (!(exponent & 1).isZero()) result *= base;
  base *= base;
  exponent >>= 1;
}
```

Feedback:
  1) We don't need kOne because `&` takes regular integers as well.
  2) `isZero` check of the result of `&` represents the intent more directly - check whether the bit is set by comparing the output of AND with zero.
  3) `exponent >>= 1` is a thing, so let's leverage it.

Reply:
Regarding 3
error: no match for ‘operator>>=’ (operand types are ‘llvm::APInt’ and ‘int’)
error: no match for ‘operator>>’ (operand types are ‘llvm::APInt’ and ‘int’)
error: no match for ‘operator>>’ (operand types are ‘llvm::APInt’ and ‘llvm::APInt’)